src/wacks4shop/wack.py

```diff
@@ -82,9 +85,16 @@ def main(db: Wackabase, dry=False):
 
     args = parser.parse_args()
 
-    # create the database folder if it doesn't exit
-    Path(DATABASE_DIR).mkdir(parents=True, exist_ok=True)
-
-    wackabase = Wackabase(DATABASE_DIR)
-    main(db=wackabase, dry=args.dry)
-    wackabase.write_success()
+    # grab the lock
+    try:
+        lock = FileLock(LOCKFILE, timeout=5)
```

Comment:
does this need to be added to each integration's wack.py?

Reply:
etsy's is generally pretty fast so we may not need to worry about it yet. I'd rather try it out on one first